system_setup/common/wsl_conf.py

```diff
@@ -0,0 +1,166 @@
+
+    :param is_advanced: boolean, True if it is WSLConfigurationAdvanced,
+                        else is WSLConfigurationBase
+    """
+    data = {}
+    conf_ref = config_adv_default if is_advanced else config_base_default
+    data = wsl_config_loader(data, "/etc/wsl.conf", conf_ref, "wsl")
+    if is_advanced:
+        data = \
+            wsl_config_loader(data, "/etc/ubuntu-wsl.conf", conf_ref, "ubuntu")
+    return data
+
+
+# TODO: remove dryrun and add root param once we write the option to .subiquity
+def wsl_config_update(config_class, is_dry_run):
+    """
+    This update the configuration file for the given class.zzd
+
+    :param config_class: WSLConfigurationBase or WSLConfigurationAdvanced
+    :param is_dry_run: boolean, True if it is a dry run
+    """
+    if is_dry_run:
+        log.debug("mimicking setting config %s", config_class)
+    temp_conf_default = {}
+    temp_confname = config_class.__str__()
+    if temp_confname.startswith("WSLConfigurationBase"):
+        temp_conf_default = config_base_default
+    elif temp_confname.startswith("WSLConfigurationAdvanced"):
+        temp_conf_default = config_adv_default
+    else:
+        raise TypeError("Invalid type name.")
+    if is_dry_run:
+        return
+
+    # update the config file
+    for config_type in temp_conf_default:
+        config_sections = temp_conf_default[config_type]
+
+        config = ConfigParser()
+        config.BasicInterpolcation = None
+
+        if config_type == "wsl":
+            conf_file = "/etc/wsl.conf"
+        elif config_type == "ubuntu":
+            conf_file = "/etc/ubuntu-wsl.conf"
+        else:
+            raise TypeError("Invalid type name " % config_type)
+
+        config.read(conf_file)
+
+        for config_section in config_sections:
+            config_settings = config_sections[config_section]
+            for config_setting in config_settings:
+                config_default_value = config_settings[config_setting]
+                config_api_name = \
+                    config_section.lower() + "_" + config_setting.lower()
+                config_value = config_class.__dict__[config_api_name]
+                # if the value for the setting is defualt value, drop it
+                if config_default_value == config_value:
+                    if config_setting in config[config_section]:
+                        config.remove_option(config_section, config_setting)
+                    # drop the section if it become empty
+                    if config[config_section] == {}:
+                        config.remove_section(config_section)
+                else:
+                    config[config_section][config_setting] = config_value
+
+        with open(conf_file + ".new", 'w') as configfile:
+            config.write(configfile)
+
+        if os.isfile(conf_file):
+            os.rename(conf_file, conf_file + ".old")
+        os.move(conf_file + ".new", conf_file)
```

Comment:
This should be os.rename(), no? Why you didn’t use the same python function both times?

Also., os.move doesn’t exit, how did it work for you and in the tests?

Reply:
This is my mistake here, sorry. This should now be fixed

Reply:
Ack, I think this meant you didn’t test your changes. Please be more careful next time, local test should pass and automated ones as well. Here none of them passed.